src/Microsoft.AspNet.Mvc.Core/Filters/FormatFilterAttribute.cs

```diff
@@ -0,0 +1,132 @@
+        /// 2. If there is a conflicting producesFilter.
+        /// </summary>
+        /// <param name="context">The <see cref="ResourceExecutingContext"/>.</param>
+        public void OnResourceExecuting([NotNull] ResourceExecutingContext context)
+        {
+            var format = GetFormat(context);
+
+            if (!string.IsNullOrEmpty(format))
+            {
+                var formatContentType = GetContentType(format, context);
+                if (formatContentType == null)
+                {
+                    // no contentType exists for the format, return 404
+                    context.Result = new HttpNotFoundResult();
+                }
+                else
+                {
+                    var responseTypeFilters = context.Filters.OfType<IApiResponseMetadataProvider>();                    
+                    var contentTypes = new List<MediaTypeHeaderValue>();
+
+                    foreach (var filter in responseTypeFilters)
+                    {
+                        filter.SetContentTypes(contentTypes);
+                    }
+
+                    if (contentTypes.Count != 0)
+                    {
+                        // If formatfilterContentType is not subset of any of the content types produced by 
+                        // IApiResponseMetadataProviders, return 404
+                        if (!contentTypes.Any(c => formatContentType.IsSubsetOf(c)))
+                        {
+                            context.Result = new HttpNotFoundResult();
+                        }
+                    }                   
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public void OnResourceExecuted([NotNull] ResourceExecutedContext context)
+        {
+
+        }
+
+        /// <summary>
+        /// Sets a Content Type on an  <see cref="ObjectResult" />  using a format value from the request.
+        /// </summary>
+        /// <param name="context">The <see cref="ResultExecutingContext"/>.</param>
+        public void OnResultExecuting([NotNull] ResultExecutingContext context)
+        {
+            var format = GetFormat(context);
+            if (!string.IsNullOrEmpty(format))
+            {
+                var objectResult = context.Result as ObjectResult;
+                if (objectResult != null)
+                {
+                    var contentType = GetContentType(format, context);
+                    Debug.Assert(contentType != null);
+                    objectResult.ContentTypes.Clear();
+                    objectResult.ContentTypes.Add(contentType);
+                }
+            }
+        }
+        
+        /// <inheritdoc />     
+        public void OnResultExecuted([NotNull] ResultExecutedContext context)
+        {
+
+        }
+
+        /// <summary>
+        /// If the current request contains format value, returns true. It means the format filter is going to execute.
```

Comment:
I have reworded it. I think it is necessary to connect that format value in the URL means that the format is active.